modules/ocf/manifests/firewall.pp

```diff
@@ -0,0 +1,15 @@
+class ocf::firewall {
+  include firewall
+
+  # One unpleasant thing about the puppetlabs-firewall module is that it
+  # calls iptables-save, which saves all iptables rules when it runs,
+  # including any transient rules added by Docker that happen to be
+  # present. These transient rules then get saved and loaded at every boot.
+  # Docker is designed to clear away or work with most existing iptables
+  # rules. However, it doesn't clear POSTROUTING rules as we'd like it to,
+  # so we have to manually ensure those don't get persisted.
+  Firewall <| |> -> exec { 'Delete Docker POSTROUTING rules in iptables persistent config':
+    command => "sed -i '/^-A POSTROUTING / d' /etc/iptables/rules.v4",
+    onlyif  => "grep '^-A POSTROUTING ' /etc/iptables/rules.v4",
+  }
```

Comment:
I really don't like the sed-ing out here and would prefer you to not do it and just live with the fact that Puppet will save the rules, knowing that eventually we'll fix that. But it's not a big deal so I leave it up to you.